engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -97,127 +98,122 @@ void postConstruct() {
             aggregationCommand = objectMapper.readValue(aggregationCommandJson, AggregationCommand.class);
         } catch (JsonProcessingException e) {
             LOGGER.error("Kafka aggregation payload parsing failed", e);
             rejectedAggregationCommandCount.increment();
-            return Uni.createFrom().nullItem();
+            return;
         }
 
         LOGGER.infof("Processing received aggregation command: %s", aggregationCommand);
         processedAggregationCommandCount.increment();
 
-        return sessionFactory.withStatelessSession(statelessSession -> {
-            return processAggregateEmailsByAggregationKey(
-                    aggregationCommand.getAggregationKey(),
-                    aggregationCommand.getStart(),
-                    aggregationCommand.getEnd(),
-                    aggregationCommand.getSubscriptionType(),
-                    // Delete on daily
-                    aggregationCommand.getSubscriptionType().equals(EmailSubscriptionType.DAILY)
-            ).onItem().ignoreAsUni();
-        })
-        .onFailure().recoverWithUni(t -> {
-            LOGGER.info("Error while processing aggregation", t);
+        try {
+            statelessSessionFactory.withSession(statelessSession -> {
+                processAggregateEmailsByAggregationKey(
+                        aggregationCommand.getAggregationKey(),
+                        aggregationCommand.getStart(),
+                        aggregationCommand.getEnd(),
+                        aggregationCommand.getSubscriptionType(),
+                        // Delete on daily
+                        aggregationCommand.getSubscriptionType().equals(EmailSubscriptionType.DAILY));
+            });
+        } catch (Exception e) {
+            LOGGER.info("Error while processing aggregation", e);
             failedAggregationCommandCount.increment();
-            return Uni.createFrom().nullItem();
-        });
+        }
     }
 
-    private Multi<Tuple2<NotificationHistory, EmailAggregationKey>> processAggregateEmailsByAggregationKey(EmailAggregationKey aggregationKey, LocalDateTime startTime, LocalDateTime endTime, EmailSubscriptionType emailSubscriptionType, boolean delete) {
+    private List<Tuple2<NotificationHistory, EmailAggregationKey>> processAggregateEmailsByAggregationKey(EmailAggregationKey aggregationKey, LocalDateTime startTime, LocalDateTime endTime, EmailSubscriptionType emailSubscriptionType, boolean delete) {
         final EmailTemplate emailTemplate = emailTemplateFactory.get(aggregationKey.getBundle(), aggregationKey.getApplication());
 
-        Multi<Tuple2<NotificationHistory, EmailAggregationKey>> doDelete = delete ?
-                emailAggregationRepository.purgeOldAggregation(aggregationKey, endTime)
-                        .onItem().transformToMulti(unused -> Multi.createFrom().empty()) :
-                Multi.createFrom().empty();
+        if (delete) {
+            // TODO NOTIF-488 To be changed
```

Comment:
What is this todo for?

Reply:
That's one of the unfinished things I'll fix tomorrow 😃

Reply:
Removed.